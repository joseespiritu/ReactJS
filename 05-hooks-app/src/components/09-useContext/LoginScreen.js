import React, { useContext } from "react";
import { UserContext } from "./UserContext";

const LoginScreen = () => {
  // 1. Obtener refenrecia al userContext
  // 2. setUser
  // {id: 132, name: 'Jose'}

  const { setUser } = useContext(UserContext);

  return (
    <div>
      <h1>LoginScreen</h1>
      <hr />
      <button
        className="btn btn-primary"
        onClick={() => setUser({ id: 132, name: "Jose" })}
      >
        Login
      </button>
    </div>
  );
};

export default LoginScreen;
