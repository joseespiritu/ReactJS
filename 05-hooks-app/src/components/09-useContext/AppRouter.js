import React from "react";
import { Routes, Route, BrowserRouter } from "react-router-dom";
import AboutScreen from "./AboutScreen";
import LoginScreen from "./LoginScreen";
import HomeScreen from "./HomeScreen";
import { NavBar } from "./NavBar";

const AppRouter = () => {
  return (
    <BrowserRouter>
      <div>
        <NavBar />
        <div className="container">
          <Routes>
            <Route path="/" element={<HomeScreen />} />
            <Route path="about" element={<AboutScreen />} />
            <Route path="login" element={<LoginScreen />} />
            <Route path="*" index element={<HomeScreen />} />
          </Routes>
        </div>
      </div>
    </BrowserRouter>
  );
};

export default AppRouter;
