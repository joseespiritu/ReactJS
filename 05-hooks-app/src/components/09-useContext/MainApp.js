import React, { useState } from "react";
import AppRouter from "./AppRouter";
import { UserContext } from "./UserContext";

const MainApp = () => {
  const [user, setUser] = useState({});

  // const user = {
  //   id: 1234,
  //   name: "jose",
  //   email: "jose@email.com",
  // };

  return (
    <UserContext.Provider
      value={{
        user,
        setUser,
      }}
    >
      <AppRouter />
    </UserContext.Provider>
  );
};

export default MainApp;
