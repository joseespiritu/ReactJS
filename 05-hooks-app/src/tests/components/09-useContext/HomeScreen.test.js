import { mount, shallow } from "enzyme";
import React from "react";
import HomeScreen from "../../../components/09-useContext/HomeScreen";
import { UserContext } from "../../../components/09-useContext/UserContext";

describe("Pruebas en <HomeScreen/>", () => {
  const user = {
    name: "jose",
    email: "jose@email.com",
  };
  const wrapper = shallow(
    <UserContext.Provider
      value={{
        user,
      }}
    >
      <HomeScreen />
    </UserContext.Provider>
  );
  test("debe mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
