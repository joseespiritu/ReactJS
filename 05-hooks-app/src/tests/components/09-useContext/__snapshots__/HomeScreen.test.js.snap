// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HomeScreen/> debe mostrarse correctamente 1`] = `<HomeScreen />`;
